src/main/java/org/jabref/gui/importer/NewEntryAction.java

```diff
@@ -53,17 +63,26 @@ public void execute() {
             return;
         }
 
-        if (type.isPresent()) {
-            jabRefFrame.getCurrentBasePanel().insertEntry(new BibEntry(type.get()));
-        } else {
-            EntryTypeView typeChoiceDialog = new EntryTypeView(jabRefFrame.getCurrentBasePanel(), dialogService, preferences);
-            EntryType selectedType = typeChoiceDialog.showAndWait().orElse(null);
+        if (fromID){
```

Comment:
Boolean parameters should not be used. This smells for a second method (or enums - see http://media.pragprog.com/titles/javacomp/split.pdf)

Reply:
I think this should actually be a completely new class.